test/rate_limiter_test.cc

```diff
@@ -207,31 +231,65 @@ TEST_F(RateLimiterTest, LinearRampingRateLimiterImplInvalidArgumentTest) {
 }
 
 TEST_F(LinearRampingRateLimiterImplTest, TimingVerificationTest) {
-  // The first acquisition timing (1000050) seems slightly off because of floating point
-  // rounding causing the first timing to fall into the next clock tick. This is
-  // a test only issue: in practice we don't have fixed microsecond step sizes,
-  // and the rate limiter computes at nanosecond precision internally. As we
-  // want to have microsecond level precision, this should be more then sufficient.
-  EXPECT_EQ(getAcquisitionTimings(5_Hz, 5s),
-            std::vector<int64_t>({1000050, 1732100, 2236100, 2645800, 3000000, 3316650, 3605600,
-                                  3873000, 4123150, 4358900, 4582600, 4795850, 5000000}));
-  EXPECT_EQ(getAcquisitionTimings(4_Hz, 2s),
-            std::vector<int64_t>({707150, 1224750, 1581150, 1870850}));
+  checkAcquisitionTimings(1_Hz, 3s);
```

Comment:
This is indeed much better for readability, but i am debating with myself whether it doesn't decrease our coverage. Could there later be a bug that would go undetected? Should we add at least one test case that verifies the exact values as before? WDYT?

Reply:
yeah this makes sense, done (f480248669da8e5a839d23fd000b1c3113731406)